admin/spec/models/check-window.spec.js

```diff
@@ -53,12 +68,100 @@ describe('check-window schema', function () {
     })
   })
 
-  it('requires a checkEndDate property', function (done) {
+  it('should require a checkEndDate property', function (done) {
     checkWindow.checkEndDate = undefined
     checkWindow.validate(error => {
       expect(error).toBeDefined()
       expect(error.errors.checkEndDate).toBeDefined()
       done()
     })
   })
+
+  describe('Method getCheckWindow', function () {
+    it('should return a check window by id', async function (done) {
+      expect(typeof CheckWindow.getCheckWindow).toBe('function')
+      sandbox
+        .mock(CheckWindow)
+        .expects('findOne')
+        .chain('exec')
+        .resolves(CheckWindowMock)
+
+      try {
+        const cw = await CheckWindow.getCheckWindow('5964fb80b42c79b8849b33f0')
+        const startDate = moment.utc('2017-08-26 00:00:01').toDate()
+        const checkStartDate = moment.utc('2017-09-10 00:00:01').toDate()
+        const checkEndDate = moment.utc('2017-09-12 00:00:01').toDate()
+
+        expect(cw.checkWindowName).toBe('Check Window Name')
+        expect(cw.startDate.toString()).toBe(startDate.toString())
+        expect(cw.checkStartDate.toString()).toBe(checkStartDate.toString())
+        expect(cw.checkEndDate.toString()).toBe(checkEndDate.toString())
+        expect(cw.isDeleted).toBe(false)
+        done()
+      } catch (err) {
+        console.log('ERROR', err)
+      }
+      done()
+    })
+
+    it('should fails', async function (done) {
+      const message = 'Failed to get check windows for form.'
+      sandbox
+        .mock(CheckWindow)
+        .expects('findOne')
+        .chain('exec')
+        .rejects(new Error(message))
+
+      try {
+        await CheckWindow.getCheckWindow('5964fb80b42c79b8849b33f0')
+        done(new Error('Failed to catch rejected promise'))
+      } catch (err) {
+        expect(err.message).toBe(message)
+      }
+      done()
+    })
+  })
+
+  describe('Method getCheckWindowsAssignedToForms', function () {
+    it('should return the expected results', async function (done) {
+      expect(typeof CheckWindow.getCheckWindowsAssignedToForms).toBe('function')
+      sandbox
+        .mock(CheckWindow)
+        .expects('find')
+        .chain('exec')
+        .resolves(CheckWindowMocks)
+
+      try {
```

Comment:
is the try catch needed here?
i would assume you would only need it in case you want to add an expectation on the catch part